openedx_external_enrollments/external_enrollments/salesforce_external_enrollment.py

```diff
@@ -166,22 +166,18 @@ def _get_program_of_interest_data(self, data, order_lines):
                     bundle_id=program.get("uuid"),
                 )
                 program_of_interest = related_program.meta
-                program_of_interest["Drupal_ID"] = "enrollment+program+{}+{}".format(
-                    openedx_user.username,
-                    request_time.strftime("%Y/%m/%d-%H:%M:%S"),
-                )
             else:
                 for line in order_lines:
                     course = self._get_course(line.get("course_id"))
                     program_of_interest = course.other_course_settings.get("salesforce_data")
                     if program_of_interest:
                         break
 
-                program_of_interest["Drupal_ID"] = "enrollment+course+{}+{}".format(
-                    openedx_user.username,
-                    request_time.strftime("%Y-%m-%d-%H:%M:%S"),
-                )
-
+            program_of_interest["Drupal_ID"] = "enrollment+{}+{}+{}".format(
+                "program" if program else "course",
+                openedx_user.username,
+                request_time.strftime("%Y/%m/%d-%H:%M:%S") if program else request_time.strftime("%Y-%m-%d-%H:%M:%S"),
```

Comment:
The request time will be the same in both cases:
```suggestion
                request_time.strftime("%Y/%m/%d-%H:%M:%S")
```
Also, the trailing comma is unnecessary since we know there won't be any other argument for the resulting string.